Options.py

```diff
@@ -466,6 +465,94 @@ def __eq__(self, other: typing.Any):
             raise TypeError(f"Can't compare {self.__class__.__name__} with {other.__class__.__name__}")
 
 
+class Bosses(TextChoice):
+    """Generic boss shuffle option that supports plando. Format expected is 'boss-location;boss-location;shuffle_type'
+    if shuffle_type is not provided in the string, will default to null. Must override can_place_boss, which
+    passes a plando boss and location. Check if the placement is valid for your game here."""
+    bosses: set = set()
+    locations: set = set()
+
+    @classmethod
+    def from_text(cls, text: str):
+        # set all of our text to lower case for name checking
+        text = text.lower()
+        if text == "random":
+            import random
+            return cls(random.choice(list(cls.options.values())))
+        for option_name, value in cls.options.items():
+            if option_name == text:
+                return cls(value)
+        cls.bosses = {boss_name.lower() for boss_name in cls.bosses}
+        cls.locations = {boss_location.lower() for boss_location in cls.locations}
```

Comment:
It does not seem right to do it here, and each time - i.e. in from_text - i think metaclass magic would be the proper place :S

Also i think it might be smart to do it as frozenset

Reply:
mmm the class doesn't actually get instantiated until after `from_text` finishes resolving which is after all the checks resolve so i'm unsure how to resolve an attribute that way.